### ISSUE TITLE
clang: Invalid address-of-packed-member warning

### ISSUE THREAD

**Issue by jstengleingithub** [llvm/llvm-project#144729]
https://godbolt.org/z/vKxY544zz

```c
typedef struct __attribute__ ((packed)) {
    char a;
    int element;
} packed_struct_type;

void somefunc (void *a, int b)
{
}

int main (int argc, char **argv)
{
    packed_struct_type packed_struct;

    somefunc(&packed_struct.element, ({3;}));
    somefunc(&packed_struct.element, 3);
} 
```

diagnostic:

```console
<source>:14:15: warning: taking address of packed member 'element' of class or structure 'packed_struct_type' may result in an unaligned pointer value [-Waddress-of-packed-member]
   14 |     somefunc(&packed_struct.element, ({3;}));
      |  
```

See an invalid address-of-packed-member warning here.  Passing the pointer to a void* parameter should not cause the warning to show, however see the warning in the case of the statement expression in the function call on line 14.
